### PR TITLE
Fix: page wrapper and routing re-org

### PR DESCRIPTION
**Pull request by Jake-Lindvall**
* Moving the router `<Switch>` into the `<PageWrapper>` component - hopefully a good home for future development and organization
* Extracting search results into their own new component, also rendered under the `/search` URL per the [spec](https://www.figma.com/file/JRTYL35e6a27C3PaP3h4xp/Kirby-UI-v2.0?node-id=182%3A2233)
* Using some fun `react-router` voodoo to allow a key press of `Enter` in the search bar to direct to the new, appropriate `/search` route 
* Small naming refactors for clarity, such as `<SearchModal>` => `<SearchButton>` for the magnifying glass in the App Bar (NOTE: if these cause unnecessary conflicts, they can be removed/reverted for now)
___
* Cleaning up and restructuring as necessary because of the above

### CODE REVIEW

**Review by davidkliewer-kirby — DISMISSED** [review]

**Review by robertwollny-pnmac — DISMISSED** [review]

File: src/Components/Scenes/SearchResults/SearchResults/SearchResults.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/styles';
+import Splash from '../../../Presentational/Splash';
+import TableSectionContainer from '../../SearchResults/TableSection/TableSection-Container';
+import TableSkeleton from '../../SearchResults/TableSkeleton/TableSkeleton';
+
+const searchResultsStyle = makeStyles(theme => ({
+  // not certain if we will want or need separate styles at the moment
+  searchResults: {}
```
Comment (SelcukEdma) [inline src/Components/Scenes/SearchResults/SearchResults/SearchResults.js]: If we are not using it we can remove it.

File: src/Components/Scenes/SearchResults/CheckBoxButton/CheckBoxButton.js [hunk 2]
```diff
@@ -16,24 +16,24 @@ const useStyles = makeStyles(theme => ({
...
   const { searchResultCopy, requestAssetsClick } = props;
+  const classes = useStyles();
   const searchResultChecked = searchResultCopy.filter(obj => obj.checked);
-  const searchResultLenght = searchResultChecked.length;
+  const searchResultLength = searchResultChecked.length;
   const buttonText =
-    searchResultLenght === 1
-      ? `Request ${searchResultLenght} Asset`
-      : searchResultLenght > 1
-      ? `Request ${searchResultLenght} Assets`
```
Comment (SelcukEdma) [inline src/Components/Scenes/SearchResults/CheckBoxButton/CheckBoxButton.js:27]: aww my misspelling :(

**Review by SelcukEdma — DISMISSED** [review]

File: src/Components/Scenes/SearchResults/SearchResults/SearchResults.js (hunk 1 above)
Comment (Jake-Lindvall) [reply src/Components/Scenes/SearchResults/SearchResults/SearchResults.js]: thank you. will remove, kind of forgot to address that haha

**Review by ghost — APPROVED** [review]
looks good to me!

**Review by lheld-pnmac — APPROVED** [review]
lgtm 👍

**Pull request merged**